src/inpod/workloadmanager.rs

```diff
@@ -186,15 +184,17 @@ impl WorkloadProxyManager {
                 Ok(()) => {
                     info!("process stream ended with eof");
                 }
+                Err(Error::ProtocolError) => {
+                    error!("protocol mismatch error while processing stream, shutting down");
+                    return
```

Comment:
do we need to bubble up an error so https://github.com/istio/ztunnel/blob/c84404d0456a1c6677966277d7f84622e655cdc7/src/app.rs#L335 hits?